dev/src/main/java/org/eclipse/codewind/intellij/ui/wizard/AddExistingProjectWizard.java

```diff
@@ -91,37 +97,80 @@ protected void doOKAction() {
         // Use the detected type if the validation page is active otherwise use the type from the project type page
         final ProjectInfo projectInfo = model.getProjectInfo();
         final ProjectTypeInfo typeInfo = model.getProjectTypeInfo();
-        ProgressManager.getInstance().runProcessWithProgressSynchronously(new Runnable() {
-            @Override
-            public void run() {
-                try {
-                    ProgressIndicator indicator = ProgressManager.getInstance().getProgressIndicator();
-
-                    // Todo: for remote, check for a push registry if Codewind style project.  See Codewind for Eclipse
-
-                    // Bind the project to the connection
-                    String path = model.getProjectPath();
-                    if (projectInfo != null && projectInfo.type.getId().equals(typeInfo.getId())) {
-                        ProjectUtil.bindProject(name, path, projectInfo.language.getId(), projectInfo.type.getId(), connection.getConid(), indicator);
-                    } else { // If the user chooses some other type instead of the detected type
-                        final ProjectSubtypeInfo subtypeInfo = model.getSubtypeInfo();
-                        String language = subtypeInfo.id;
-                        // call validate again with type and subtype hint
-                        // allows it to run extension commands if defined for that type and subtype
-                        if (subtypeInfo != null) {
+        // Todo: for remote, check for a push registry if Codewind style project.  See Codewind for Eclipse
+
+        // Bind the project to the connection
+        String path = model.getProjectPath();
+        bindException = null;
+        if (projectInfo != null && projectInfo.type.getId().equals(typeInfo.getId())) {
+            continueWithBind(name, path, projectInfo.language.getId(), typeInfo);
+        } else { // If the user chooses some other type instead of the detected type
+            final ProjectSubtypeInfo subtypeInfo = model.getSubtypeInfo();
+            String language = subtypeInfo.id;
```

Comment:
There's a check for `subtypeInfo` being null a few lines down.  If `subtypeInfo` can be null, the declaration of `language` should be moved inside that null-check.  Otherwise the null-check should be removed.